php/libraries/SubprojectController.class.inc

```diff
@@ -0,0 +1,252 @@
+     *
+     * @return \LORIS\Subproject
+     */
+    public function getSubprojectByID(int $subprojectID) : \LORIS\Subproject
+    {
+        $subprojectTitle = $this->database->pSelectOne(
+            "SELECT
+              title
+            FROM
+              subproject
+            WHERE SubprojectID = :id
+            ",
+            ['id' => $subprojectID]
+        );
+
+        if (empty($subprojectTitle)) {
+            throw new \LorisException("No subproject with ID $subprojectID found.");
+        }
+
+        return $this->getSubprojectByTitle($subprojectTitle);
+    }
+
+    /**
+     * Create a new subproject and insert it into the database.
+     *
+     * @param \LORIS\Subproject $subproject A subproject object without the ID
+     *
+     * @return \LORIS\Subproject The created subproject with it's new ID
+     */
+    public function createSubproject(\LORIS\Subproject $subproject)
+    : \LORIS\Subproject
+    {
+        if (!empty($subproject->getID())) {
+            throw new \LorisException(
+                "An ID can not be supplied when creating a new subproject, the ID
+                will be generated by the database."
+            );
+        }
+        $subprojectExists = $this->subprojectExists($subproject->getTitle());
+
+        if (!$subprojectExists) {
+            throw new \LorisException(
+                "A subproject with the same title ".
+                $subproject->getTitle()." already exists"
+            );
+        }
+
+        // Update subproject table
+        $this->database->insert(
+            'subproject',
+            [
+                'title'             => $subproject->getTitle(),
+                'useEDC'            => is_null($subproject->getUseEDC())
+                    ?: ($subproject->getUseEDC() ? 1 : 0),
+                'WindowDifference'  => $subproject->getWindowDifference(),
+                'RecruitmentTarget' => $subproject->getRecruitmentTarget(),
+            ]
+        );
+
+        return $this->getSubprojectByTitle($subproject->getTitle());
+    }
+
+    /**
+     * Update an existing subproject in the database.
+     *
+     * @param \LORIS\Subproject $subproject The subproject object to update
+     *
+     * @return \LORIS\Subproject The updated subproject
+     */
+    public function updateSubproject(\LORIS\Subproject $subproject)
+    : \LORIS\Subproject
+    {
+        // Update subproject table
+        $this->database->update(
+            'subproject',
+            [
+                'title'             => $subproject->getTitle(),
+                'useEDC'            => is_null($subproject->getUseEDC())
+                    ?: ($subproject->getUseEDC() ? 1 : 0),
+                'WindowDifference'  => $subproject->getWindowDifference(),
+                'RecruitmentTarget' => $subproject->getRecruitmentTarget(),
+            ],
+            ['title'  => $subproject->getTitle()]
+        );
+
```

Comment:
The `$_instances` variable should be updated as well.

Reply:
probably gonna remove it